src/Context/SoapContextReader.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * @author Sergii Bondarenko, <sb@firstvector.org>
+ */
+namespace Behat\SoapExtension\Context;
+
+use Behat\Testwork\Call\Callee;
+use Behat\Testwork\Environment\Environment;
+use Behat\Testwork\Environment\Reader\EnvironmentReader;
+use Behat\Behat\Context\Reader\ContextReader;
+use Behat\Behat\Context\Environment\ContextEnvironment;
+use Behat\Behat\Context\Environment\UninitializedContextEnvironment;
+
+/**
+ * Class SoapContextReader.
+ *
+ * @package Behat\SoapExtension\Context
+ */
+class SoapContextReader implements EnvironmentReader
+{
+    /**
+     * @var ContextReader[]
+     */
+    private $contextReaders = [];
+    /**
+     * @var Callee[]
+     */
+    private $callees = [];
+
+    /**
+     * Register context loaders.
+     *
+     * @see SoapExtension::process()
+     *
+     * @param ContextReader[] $contextReaders
+     */
+    public function __construct(array $contextReaders)
+    {
+        $this->contextReaders = $contextReaders;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supportsEnvironment(Environment $environment)
+    {
+        return $environment instanceof ContextEnvironment;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function readEnvironmentCallees(Environment $environment)
+    {
+        if ($environment instanceof UninitializedContextEnvironment) {
```

Comment:
I'm not very familiar with all this magic context initialization, but i think you're trying to include any context that we had created in the SoapExtension to be automatically included and loaded by Behat.

While this seems reasonable, I have a question, what if someone wants to use one context only, not to load steps from other contexts. Currently Drupal extension for example includes all contexts one by one in behat.yml file.
```
- Drupal\DrupalExtension\Context\MinkContext: ~
- Drupal\DrupalExtension\Context\DrupalContext: ~
- Drupal\DrupalExtension\Context\MarkupContext: ~
- Drupal\DrupalExtension\Context\MessageContext: ~
```

Considering the fact that we only have one Context to include here, and probably not much else will appear in the future, I'm thinking what value the autoloading Context will bring, other than no additional step required in behat.yml.

Ideas?